dev/release/verify-release-candidate.bat

```diff
@@ -140,7 +140,7 @@ set PYARROW_WITH_PARQUET=1
 set PYARROW_WITH_PARQUET_ENCRYPTION=1
 set PYARROW_WITH_DATASET=1
 set PYARROW_TEST_CYTHON=OFF
-python setup.py build_ext --inplace --bundle-arrow-cpp bdist_wheel || exit /B 1
+python -m build --wheel --config-settings=--bundle-arrow-cpp --no-isolation . || exit /B 1
```

Comment:
Note this ugly `--config-settings` is why it's nice to have the options configurable via environment variables as well ;)

Reply:
Yea I too find this pretty verbose. We can go back to using -C instead of --config-settings if we want that succinctness again

Reply:
Or simply `set PYARROW_BUNDLE_ARROW_CPP=1`.

Reply:
That works too. The documented approach for this currently the command line argument, but it appears that more generally in the Python documentation that env vars are the documented approach. 

Maybe its worth removing the CLI arguments from setuptools if the environment variable approach is the one we want to stick to (in a separate PR?)

Reply:
We can deprecate them (in a separate PR indeed) but we can't remove them immediately as it would break user builds.